api/proto/external.proto

```diff
@@ -179,3 +188,26 @@ message TxHash {
     // Hash of a single transaction.
     bytes hash = 1;
 }
+
+// Given to the recipient of a transaction output by the sender so that the
+// recipient may verify that the other party is indeed the sender.
+//
+// Often given to the recipient before the transaction is finalized so that
+// the recipient may know to anticipate the arrival of a transaction output,
+// as well as know who it's from, when to consider it as having surpassed
+// the tombstone block, and the expected amount of the output.
+message Receipt {
+    // Public key of the TxOut.
+    CompressedRistretto public_key = 1;
+
+    // Confirmation number of the TxOut.
+    TxOutConfirmationNumber confirmation = 2;
+
+    // Tombstone block of the Tx that produced the TxOut.
+    // Note: This value is self-reported by the sender and is unverifiable.
+    uint64 tombstone_block = 3;
+
+    // Amount of the TxOut.
+    // Note: This value is self-reported by the sender and is unverifiable.
+    Amount amount = 4;
```

Comment:
This feels weird to me. The sender can communicate the amount, if they with to, out-of-band using the same medium they use to communicate the actual Receipt. I don't think we need to encourage this, but if others have a need for it I can live with it.

Reply:
I agree that the sender can always communicate the `Amount` out-of-band if they wish to. However, the same could be said of the tombstone block. I think it comes down to defining what the `Receipt` represents.

1) If the `Receipt` is meant solely as proof that the sender sent a particular `TxOut`, which the recipient would use after they've received the `TxOut` in question, then only the confirmation number and an identifier are needed (`TxOut` public key or `TxOut` global index would suffice).
2) If the `Receipt` is meant as a soft promise that the sender has submitted a `Tx` to consensus where one of the outputs of the `Tx` is addressed to the recipient, then it becomes a question of what might be useful for the recipient, since no statements about the ledger can be verifiably asserted until the recipient confirms that the `TxOut` is actually in the ledger, after which the only relevant field in the `Receipt` is the `TxOutConfirmationNumber`.

I propose that the `Receipt` represents the second option (which is a superset of the first option).

Given this, if the recipient is taking any action before the `TxOut` is confirmed in the ledger, then the recipient is operating on some level of trust at this point if they accept the premise that a `Tx` is in motion, and it then becomes a question of what level of trust does the recipient confer in which pieces of information, and what actions, if any, does the recipient take.
 
I contend that the `Amount` is useful for the recipient so that they may know the amount of the output, as reported by the sender. The recipient is free to trust it as they see fit. However, by create a standard using protobuf, we're ensuring that the same information is available across platforms for consideration by the recipient.

Reply:
Can't the recipient just decrypt the TxOut, and get the amount that way? Or is this like an optimization to avoid that?

Reply:
@garbageslam decrypt the `TxOut`, meaning if the `TxOut` was included in the `Receipt` instead? Yeah, they could definitely do that. This is just basically removing the `e_account_hint` and `target_key` fields, since neither of them are useful in this situation.

Reply:
I see, I didn't notice that the TxOut isn't in the receipt

Reply:
Thanks for the detailed explanation of your thinking behind this @kylefleming